common/config/loader.go

```diff
@@ -90,14 +93,28 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 	// TODO: remove log dependency.
 	stdlog.Printf("Loading config files=%v\n", files)
 
+	templateFuncs := sprig.FuncMap()
+
 	for _, f := range files {
 		// This is tagged nosec because the file names being read are for config files that are not user supplied
 		// #nosec
 		data, err := os.ReadFile(f)
 		if err != nil {
 			return err
 		}
-		err = yaml.Unmarshal(data, config)
+
+		tpl, err := template.New("config").Funcs(templateFuncs).Parse(string(data))
+		if err != nil {
+			return err
```

Comment:
Does it clear where error is coming from in case if there is template syntax error? I am wondering if you should wrap it with something like "template parsing error: %w" and "template execution error: %w" bellow. Just to make it more explicit where to look.